tools/export/makefile/__init__.py

```diff
@@ -145,6 +146,12 @@ def generate(self):
                 new_asm_flags.append(flag)
         ctx['asm_flags'] = new_asm_flags
 
+        # If there is a target-specific exporter function, call it now.
+        if self.TARGET_EXPORTER:
+            print ("Calling target exporter...")
+            self.TARGET_EXPORTER(self, ctx)
```

Comment:
As this is the only exporter that uses this "feature" I'm not convince that it's really needed. Could you explain why Makefiles are the only exporter that need to merge the Hex file? Further, https://github.com/ARMmbed/mbed-os/pull/8491/files#diff-a35f26ab826f25fd621856dc9d866598R94 and https://github.com/ARMmbed/mbed-os/pull/8491/files#diff-a35f26ab826f25fd621856dc9d866598R95 indicate that there will be overlapping regions in these merged hex files. This will cause the Makefile exporter to always export a non-functional Makefile for your target.